File: bindings/nodejs/types/payloads/transactionPayload.ts
// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0
import type { TransactionEssence } from '../transactionEssences';
import type { TypeBase } from '../typeBase';
import type { UnlockBlock } from '../unlockBlocks';

/**
 * The global type for the payload.
 */
export const TRANSACTION_PAYLOAD_TYPE = 6;

/**
 * Transaction payload.
 */
export interface TransactionPayload extends TypeBase<6> {
    /**
     * The index name.
     */
    essence: TransactionEssence;
    /**
     * The unlock blocks.
     */
    unlockBlocks: UnlockBlock;
}

File: bindings/java/iota-client-java/src/main/java/org/iota/types/TransactionPayload.java
// Copyright 2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package org.iota.types;

import com.google.gson.JsonObject;

public class TransactionPayload extends BlockPayload {

    public TransactionPayload(JsonObject jsonObject) {
        super(jsonObject);
    }

    public TransactionPayload(String jsonObject) {
        super(jsonObject);
    }

}